pyuvdata/calfits.py

```diff
@@ -40,6 +40,28 @@ def write_calfits(self, filename, spoof_nonessential=False,
         if run_check:
             self.check(run_check_acceptability=run_check_acceptability)
 
+        if self.Nfreqs > 1:
+            freq_spacing = self.freq_array[0, 1:] - self.freq_array[0, :-1]
+            if not np.isclose(np.min(freq_spacing), np.max(freq_spacing),
+                              rtol=self._freq_array.tols[0], atol=self._freq_array.tols[1]):
+                raise ValueError('The frequencies are not evenly spaced (probably '
+                                 'because of a select operation). The calfits format '
+                                 'does not support unevenly spaced frequencies.')
+            freq_spacing = freq_spacing[0]
+        else:
+            freq_spacing = self.channel_width  # or should this be the length of the frequency range?
```

Comment:
I think this should stay as is. The frequency range is the range of frequencies where the solutions apply. Ditto with the time_spacing on line 63.